tests/syslog/test_logrotate.py

```diff
@@ -0,0 +1,211 @@
+
+    config_reload(duthost, safe_reload=True)
+
+    logger.info('Restart logrotate-config service')
+    duthost.shell('sudo service logrotate-config restart')
+
+
+def get_var_log_size(duthost):
+    """
+    Check the size of /var/log folder
+    :param duthost: DUT host object
+    :return: size value
+    """
+    size = duthost.shell("sudo df -k /var/log | sed -n 2p | awk '{ print $2 }'")['stdout']
+    return int(size)
+
+
+def get_syslog_file_count(duthost):
+    """
+    Check the rotated syslog file number
+    :param duthost: DUT host object
+    :return: file number value
+    """
+    logger.info('Check rotated syslog file number')
+    num = duthost.shell('sudo ls -l /var/log | grep -E "syslog\\.[0-9]{1,4}[\\.gz]{0,1}" | wc -l')['stdout']
+    logger.debug('There are {} rotated syslog files'.format(num))
+    return int(num)
+
+
+def create_temp_syslog_file(duthost, size):
+    """
+    Create a temp syslog file with specific size and
+    :param duthost: DUT host object
+    :param size: file size in MB or KB
+    """
+    logger.info('Create a temp syslog file as {}'.format(size))
+    duthost.shell('sudo fallocate -l {} /var/log/syslog'.format(size))
+
+
+def run_logrotate(duthost, force=False):
+    """
+    Run logrotate command
+    :param duthost: DUT host object
+    :param force: force logrotate run immediately even the syslog size is very small, value is True or False
+    """
+    if force:
+        logger.debug('Make sure there is no big /var/log/syslog exist by forcing execute logrotate')
+        cmd = 'sudo /usr/sbin/logrotate -f /etc/logrotate.conf > /dev/null 2>&1'
+    else:
+        cmd = 'sudo /usr/sbin/logrotate /etc/logrotate.conf > /dev/null 2>&1'
+    logger.info('Run logrotate command: {}'.format(cmd))
+    duthost.shell(cmd)
+
+
+def multiply_with_unit(logrotate_threshold, num):
+    """
+    Multiply logrotate_threshold with number, and return the value
+    Such as '1024K' * 0.5, return '512K'
+    :param logrotate_threshold: string type threshold value with unit, such as '1024K'
+    :param num: the number need to multiply with
+    :return: value with unit, such as '512K'
+    """
+    return str(int(logrotate_threshold[:-1]) * num) + logrotate_threshold[-1]
+
+
+def validate_logrotate_function(duthost, logrotate_threshold):
+    """
+    Validate logrotate function
+    :param duthost: DUT host object
+    :param logrotate_threshold: logrotate threshold, such as 16M or 1024K
+    """
+    with allure.step('Run logrotate with force option to prepare clean syslog environment'):
+        run_logrotate(duthost, force=True)
+
+    with allure.step('There should be no logrotate process when rsyslog size is smaller than threshold {}'.format(logrotate_threshold)):
+        syslog_number_origin = get_syslog_file_count(duthost)
+        logger.info('There are {} syslog gz files'.format(syslog_number_origin))
+        create_temp_syslog_file(duthost, multiply_with_unit(logrotate_threshold, 0.9))
```

Comment:
What will happen to the existing syslog file? I believe logrotate force will initially move it out right?

Reply:
The existing syslog file would be backup.